src/transaction/locator_sr.c

```
@@ -13937,6 +13938,162 @@ has_errors_filtered_for_insert (std::vector<int> error_filter_array)
 }
 // *INDENT-ON*
 
+/* locator_put_classname_entry () - Put a new entry in the locator_Mht_classnames, used in PTS replicator (atomic_replicator)
+ *
+ * return	     : void
+ * thread_p (in)     : thread entry
+ * classname (in)    : classname to be inserted
+ * classoid (in)     : class oid of the classname
+ */
+void
+locator_put_classname_entry (THREAD_ENTRY * thread_p, const char *classname, const OID * classoid)
```

What if these new functions fail?
Or must they never fail?
I think we need to return the error and - at least - be able to debug log the error in the PTS replicator logic.

---

I return the error, and add debug logging